src/cata_tiles.cpp

```diff
@@ -1304,6 +1304,11 @@ void cata_tiles::draw( const point &dest, const tripoint &center, int width, int
     const int max_col = top_any_tile_range.p_max.x;
     const int min_row = bottom_any_tile_range.p_min.y;
     const int max_row = top_any_tile_range.p_max.y;
+    //invalidate draw_points_cache if viewport dimensions have changed
+    if( min_col != here.prev_min_col || max_col != here.prev_max_col || min_row != here.prev_min_row ||
+        max_row != here.prev_max_row ) {
+        here.draw_points_cache_dirty = true;
```

Comment:
It's better to call the public member function `cata_tiles::set_draw_cache_dirty()` rather than assigning the member variable directly.